scripts/script.py

```diff
@@ -2,7 +2,36 @@
 import asyncio
 import json
 from datasets import Dataset
+from utils import transform_to_jsonl
 import jsonlines
+from argparse import ArgumentParser
+
+
+def argument_parser():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--input_file",
+        type=str,
+        default="word.txt",
+        help="Input file containing the list of words to fetch",
+    )
+    parser.add_argument(
+        "--output_file",
+        type=str,
+        default="../data/dataset.jsonl",
+        help="Output file to store the fetched data",
+    )
+    parser.add_argument(
+        "--delay", type=int, default=1, help="Delay between each request in seconds"
+    )
+    parser.add_argument(
+        "--push_to_hub", type=bool, default=True, help="Push the dataset to the hub"
```

Comment:
1. **Incorrect Type for `--push_to_hub`**: The `--push_to_hub` argument currently uses `type=bool`, which is not appropriate. It forces the user to specify `True` or `False`, whereas simply including or omitting the argument should suffice to enable or disable this option.

2. **Default Pushing**: With `default=True`, the dataset is pushed to Hugging Face even if the user doesn't explicitly want this, potentially leading to errors if the token is not provided.

### Proposed Solution:

1. **Replace `type=bool` with `action='store_true'`**: This allows the `--push_to_hub` option to be activated simply by specifying the argument in the command line, without passing a value.

2. **Disable by Default**: Using `action='store_true'` makes the argument disabled by default, preventing the automatic pushing of the dataset.

Reply:
Hello @dofbi   , you are welcome! Your suggestion is perfect! Everything is okay on my side 👍  !